[ENG-DOCS-1424] Improve discoverability of company layer (#17509)
docker/docs: docker-hub/onboarding-faqs.md

A **Team** is a group of Docker users that belong to an organization. An organization can have multiple teams. When you first create an organization, you’ll see that you have a team, the owners team, with a single member. An organization owner can then create new teams and add members to an existing team using Docker IDs or email address and by selecting a team the user should be part of. [Learn more](manage-a-team.md).

### What's a company?

> A **Company** is a management layer that centralizes administration of multiple organizations. Administrators can add organizations with a Docker Business subscription to a company and configure settings for all organizations under the company. [Learn more](creating-companies.md).

Reviewer: I see that this page uses a generic "Learn more" CTA for link texts - this is something we try to avoid. Would you mind fixing the other occurrences as well?

```suggestion
A **Company** is a management layer that centralizes administration of multiple organizations. Administrators can add organizations with a Docker Business subscription to a company and configure settings for all organizations under the company. See [Set up your company](creating-companies.md).
```

Reviewer: note: not sure about which link text to actually use for this page 🙈 The page title is "Overview", so I used the section heading here. I can see that you used "Company layer" for the link text in `subscription/details.md`. Maybe we want to use that as a section heading in the TOC?

Author: Definitely, will go through and switch the instances of **Learn more** over to the descriptive format. Thanks, I thought I saw something about this in the style guide about this phrase but then they existed in this doc so good clarification 👍

Author: I *think* Set up your company works, since that page includes all the links to the other company-related pages. It sounds like company layer was an agreed-upon term for the feature especially in early access but is more like a description of how it works than official name.
